A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,432 @@
+  support for `weighted_clusters`, they must upgrade their clients before
+  they can start using this feature.  (We might be able to ameliorate
+  this by adding support for route matchers at the same time as we
+  add `weighted_clusters` support, in which case the final route in the
+  list can continue to use the `cluster` field but earlier routes can
+  use `weighted_clusters`.)
+
+#### CDS
+
+Once the cluster name is obtained from the __VirtualHost__ proto, the gRPC
+client will make a CDS request asking for that specific cluster name.
+
+Because we are requesting one specific resource by name, the CDS response
+should include exactly one `Cluster`.  However, many existing xDS servers
+do not support requesting one specific resource in CDS, so the gRPC client
+must be tolerant of servers that may ignore the requested resource name;
+if the server returns multiple resources, the client will look for the one
+with the name it asked for, and it will ignore the rest of the entries.
+Note that this means that the resource returned by the xDS server must
+have exactly the name specified by the client.
+
+#### Cluster Proto
+
+The `Cluster` proto must have the following fields set:
+
+- The `type` field must be set to EDS.
+- In the `eds_cluster_config` field, the `eds_config` field must be set to
+  indicate to use EDS (and the __ConfigSource__ proto for the EDS server
+  must indicate to use ADS).  If the `service_name` field is set, that value
+  will be used for the EDS request instead of the cluster name.
+- The `lb_policy` field must be set to `ROUND_ROBIN`.
+- If the `lrs_server` field is set, it must have its `self` field set, in
+  which case the client should use LRS for load reporting.  Otherwise
+  (the `lrs_server` field is not set), LRS load reporting will be disabled.
+
+#### EDS
+
+After getting the `Cluster` proto, the gRPC client will make an EDS
+request for a specific resource name, using either the cluster name or
+the value of the `eds_cluster_config.service_name` field in the
+`Cluster` proto.
+
+#### ClusterLoadAssignment Proto
+
+The `ClusterLoadAssignment` proto must have the following fields set:
+
+- The `endpoints` field must contain at least one entry.  In each entry:
+  - If the `load_balancing_weight` field is unset, the `endpoints` entry
+    is skipped; otherwise, the value is used for weighted locality picking.
+  - The `priority` field must be set.
+  - The `lb_endpoints` field must contain at least one entry.  In each entry:
+    - If the `health_status` field has any value other than HEALTHY or
+      UNKNOWN, the entry will be ignored.
+    - The `endpoint` field must be set.  Inside of it:
+      - The `address` field must be set.
+- The `policy.drop_overloads` field may be set.
+
+FIXME: Is this true?  Need to resolve internal discussion and decide
+whether to remove this before merging this gRFC.
+- The `policy.disable_overprovisioning` field must be set to true.
```

Comment:
I feel this is a good candidate for client feature capabilities.

Reply:
We have decided to do this via a client capability.  I added the capability to the xDS API docs in envoyproxy/envoy#10136, and I've updated this gRFC to reflect that choice.